src/app/common/util.service.js

```diff
@@ -225,6 +225,28 @@ module.exports.getSort = function (
 	return [{ property: sort, direction: dir }];
 };
 
+/**
+ * Get the sort provided by the user, if there is one.
+ *
+ * @param {Object} queryParams
+ * @param {string} [defaultSort=undefined] (optional)
+ * @param {string} [defaultDir=ASC] (optional) default: ASC
+ * @returns {Object|null}
+ */
+module.exports.getSortObj = function (
+	queryParams,
+	defaultSort,
+	defaultDir = 'ASC'
+) {
+	const sort = _.get(queryParams, 'sort', defaultSort);
+	const dir = _.get(queryParams, 'dir', defaultDir);
+	if (!sort) {
+		return null;
+	}
+
+	return { [sort]: dir };
+};
+
```

Comment:
Currently, we parse the sort options into an array format, that the paging-search plugin than converts into the object format that mongo expects.  It doesn't seem like we really need the array format so this function parses the sort options directly into the format that mongo expects.  Planning to leaving the old one for now for backward compatibility purposes, but would like to remove eventually.

Reply:
This looks like a really awesome refactor! Checks out locally. It looks like ts is freaking out a bit on the test files as it can't quite figure out that the plugin makes the proper methods available on the schema under test, but I don't know if there's really a worthwhile way of fixing that...